packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -524,53 +567,12 @@ class _MenuAnchorState extends State<MenuAnchor> {
     assert(_debugMenuInfo(
         'Opening $this at ${position ?? Offset.zero} with alignment offset ${widget.alignmentOffset ?? Offset.zero}'));
     _parent?._closeChildren(); // Close all siblings.
-    assert(_overlayEntry == null);
+    assert(!_overlayController.isShowing);
 
-    final BuildContext outerContext = context;
     _parent?._childChangedOpenState();
-    setState(() {
-      _overlayEntry = OverlayEntry(
-        builder: (BuildContext context) {
-          final OverlayState overlay = Overlay.of(outerContext);
-          return Positioned.directional(
-            textDirection: Directionality.of(outerContext),
-            top: 0,
-            start: 0,
-            child: Directionality(
-              textDirection: Directionality.of(outerContext),
-              child: InheritedTheme.captureAll(
-                // Copy all the themes from the supplied outer context to the
-                // overlay.
-                outerContext,
-                _MenuAnchorScope(
-                  // Re-advertize the anchor here in the overlay, since
-                  // otherwise a search for the anchor by descendants won't find
-                  // it.
-                  anchorKey: _anchorKey,
-                  anchor: this,
-                  isOpen: _isOpen,
-                  child: _Submenu(
-                    anchor: this,
-                    menuStyle: widget.style,
-                    alignmentOffset: widget.alignmentOffset ?? Offset.zero,
-                    menuPosition: position,
-                    clipBehavior: widget.clipBehavior,
-                    menuChildren: widget.menuChildren,
-                    crossAxisUnconstrained: widget.crossAxisUnconstrained,
-                  ),
-                ),
-                to: overlay.context,
-              ),
-            ),
-          );
-        },
-      );
-    });
+    _menuPosition = position;
```

Comment:
No, because the position is the position of the menu that it's about to show, and:
 1) the anchor doesn't have anything to update when that changes, only the overlay does and
 2) it's currently not showing, so showing it will force the redraw of the child menu anyhow.

Reply:
Ah looks like it's captured by the overlay portal's builder closure. So I guess when show() is called the up-to-date position will be used.